### DISCUSSION TITLE
how to bypass firejail when troubleshooting programs?

### DISCUSSION THREAD

**Discussion by danielkrajnik** [netblue30/firejail#4799] [Q&A] [answered]
When troubleshooting programs it would be great to once bypass firejail to check if it doesn't block something it shouldn't. Is there an easy way to do it? At the moment I can't even find where the original program is located, because all the links in /usr/bin/local point to firejail.

**Comment by rusty-snake** [comment, marked as answer]
In the most cases it is enough to start a program by it's full path (after closing all open instances if the program has instance management). Usually this is `/usr/bin/<PROGRAM>`, but for program where this is not the cases you can use `which -a <PROGRAM>`.

In rare cases you need to remove the symlink (`sudo rm /usr/local/bin/<PROGRAM>`) or all symlinks.

**Reply by danielkrajnik** [reply]
EDIT: just noticed the `which -a` command - that's an even better solution

perfect, thanks a lot. Deleting the symlink did the trick. It can be always rebuilt with `firecfg` and lets you for example run `which` on the command to find out where its original executable was located.

It's useful, because commenting out the program in `/usr/lib/x86_64-linux-gnu/firejail/firecfg.config` and running `firecfg` doesn't seem to release the sandbox. Your solution is quick and simple.

**Reply by kmk3** [reply]
Nice, I didn't know about `which -a` either.

**Reply by danielkrajnik** [reply]
I found that firejail also has a `--noprofile` option, which seems not to load any profile files. This might be an even easier way to troubleshoot sandboxed programs.
